docker/Dockerfile.rocm-complete

```
@@ -0,0 +1,68 @@
+# Docker container for running DGL on AMD GPUs with ROCM, including DGL itself.
+
+# TODO: Rebuilding DGL is slow. Consider multistage builds and shared cache
+# mounts.
+
+FROM gcmn/dgl:rocm-deps-only
```

I'd suggest letting the base image be dynamic like this, so one can modfy the deps image and use it without also modifying the rocm-complete image. Looking at the commits, it looks like you originally did that and then reverted. Is there a reason for that?

```suggestion
ARG DEPS_IMAGE=gcmn/dgl:rocm-deps-only
FROM ${BASE_IMAGE
```

---

Yeah it turns out there's now a built-in way to do that with docker as the `FROM` images are treated as named build contexts https://docs.docker.com/build/concepts/context/#example-dynamic-builds-with-custom-dependencies